src/ntfs/write.rs

```diff
@@ -239,181 +215,250 @@ struct Fares<'a> {
     ticket_use_restrictions: &'a Collection<TicketUseRestriction>,
 }
 
-fn has_constraints(ticket_use: &TicketUse) -> bool {
-    ticket_use
-        .max_transfers
-        .filter(|&mt| mt != 0)
-        .or(ticket_use.boarding_time_limit)
-        .or(ticket_use.alighting_time_limit)
-        .is_some()
-}
-
-fn get_prices<'a>(
-    ticket_prices: &'a Collection<TicketPrice>,
-    ticket_id: &str,
-) -> Vec<&'a TicketPrice> {
-    ticket_prices
+fn extract_perimeter_for_ticket_use<'id, 'p>(
+    ticket_use_id: &'id str,
+    ticket_use_perimeters: &'p Collection<TicketUsePerimeter>,
+) -> Result<(Vec<&'p str>, Vec<&'p str>, Vec<&'p str>)> {
```

Comment:
Just wondering how to make this return value more explicit (a tuple of 3 `Vec` of references on `str` does not make it very clear about what it is).

I'm wondering if it would be possible to return something like `Result<Vec<Perimeter>>` where `Perimeter` would be something like.

```rust
enum Perimeter {
  IncludeNetwork(String),
  IncludeLine(String),
  ExcludeLine(String),
}
```

Another more simple trick but that helps self-documenting would be to do something like

```rust
type IncludedNetworks<'p> = Vec<&'p str>;
type IncludedLines<'p> = Vec<&'p str>;
type ExcludedLines<'p> = Vec<&'p str>;
```
and then your function would return a type of more explicit types. This second solution is only aliases on names.

Reply:
Good idea, I added this struct as a return type 
```rust
struct Perimeter<'p> {
    included_networks : Vec<&'p str>,
    included_lines : Vec<&'p str>,
    excluded_lines : Vec<&'p str>
}
```